trade-finance-manager-ui/server/controllers/helpers/deal-cancellation-enabled.helper.test.ts

```diff
@@ -0,0 +1,12 @@
+import { DEAL_SUBMISSION_TYPE } from '@ukef/dtfs2-common';
+import { dealCancellationEnabled } from './deal-cancellation-enabled.helper';
+
+describe('dealCancellationEnabled', () => {
+  describe('when `FF_TFM_FACILITY_END_DATE_ENABLED` is set to false', () => {
+    it.each([DEAL_SUBMISSION_TYPE.MIA, DEAL_SUBMISSION_TYPE.MIN, DEAL_SUBMISSION_TYPE.AIN])('should return false if the deal type is %s', (type) => {
+      const result = dealCancellationEnabled(type);
+
+      expect(result).toEqual(false);
+    });
+  });
+});
```

Comment:
Bit confused about the 2x tests:

```ts
deal-cancellation-enabled.helper.test.ts
deal-cancellation-enabled.helper.ff-test.ts
```

Should we combine them into one?

Reply:
Good question @ttbarnes 
`deal-cancellation-enabled.helper.ff-test.ts` runs on pipeline with FF set to `true` (future state of the system)
`deal-cancellation-enabled.helper.test.ts` runs on pipeline with FF set to `false` or not set all (current state of the system)

due to above they can't be combined, as we are running two different states concurrently.

Reply:
Makes sense, thanks @abhi-markan however I think it could be made clearer by renaming both tests to the following:

```
deal-cancellation-enabled.helper.ff-enabeld.test.ts
deal-cancellation-enabled.helper.ff-disabled.test.ts
```

Or maybe

```
deal-cancellation-enabled.helper/ff-enabled.test.ts
deal-cancellation-enabled.helper/ff-disabled.test.ts
```

Perhaps this should be in a different PR so we can progress this PR and update all FF unit tests to have the same approach

Reply:
Agree with leaving this to a separate PR as that would require changing some of the test set-up I think

Reply:
Do we have a ticket for this or could this be done as part of @MarRobSoftwire 's feature flag test PR?

Reply:
Can I leave this with you @abhi-markan to decide if you want to implement this change in another ticket or not? I think I'm happy with the existing naming as long as we have some documentation about it somewhere so it's clear.

Reply:
> Do we have a ticket for this or could this be done as part of @MarRobSoftwire 's feature flag test PR?

As this is adding new features not present in Marks PR, I'd like to avoid it please. Also as this is a change in how we deal with feature flags in their whole, I think this is a bit of scope creep and we should capture this in a properly pointed and planned ticket if possible.

Reply:
Just a note that we should start to write down these feature flag concepts and schedule a meeting to talk about it -- I'll try and get some time cleared with Tom to write this up as as discussed in the office it would be incredibly beneficial to have a technical decision register to refer to

Reply:
Sounds good @AlexBramhill . Do you want to have a session before creating a ticket?

I just think the naming of the FF unit tests are confusing so would like to rename them https://github.com/UK-Export-Finance/dtfs2/pull/3458#discussion_r1756771561

Reply:
Great idea, I think this is a good example of the need for said meeting. Will need to run by Tom, which we can do in Monday standup :)

Reply:
I've added DTFS2-7371 to the backlog.

Resolving